### PR TITLE
Fix Pivot Position

### PR DESCRIPTION
**Pull request by realjoshuau**
This PR rewrites the `PivotSetAngle` class to use the Position ControlRequests, instead of using the PID output as voltage.
It also includes a failsafe to test if the motor has been stalled (based on current limits and velocity) and stops the PivotSetAngle command accordingly.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/pivot/PivotIntake.java [hunk 1]
```diff
@@ -19,19 +20,23 @@
...
 
 public class PivotIntake extends SubsystemBase implements Loggable {
 
   private TalonFX pivotMotor;
   private TalonFXSimState pivotMotorSim;
   private SingleJointedArmSim pivotModel;
-  private double offsetDeg;
 
-  final VelocityVoltage velocityRequest = new VelocityVoltage(0).withEnableFOC(true).withSlot(0);
+  final VelocityVoltage velocityRequest =
```
Comment (ChaoticChaosTheGreat) [inline src/main/java/frc/robot/subsystems/pivot/PivotIntake.java:36]: Doesn't seem to be used... Any reason we have it?

Comment (realjoshuau) [reply src/main/java/frc/robot/subsystems/pivot/PivotIntake.java]: Fixed.

File: src/main/java/frc/robot/subsystems/pivot/commands/PivotCalibrate.java [hunk 2]
```diff
@@ -7,64 +7,66 @@
...
+  private final PivotIntake pivot;
   private final int position;
-  private double prevCurrent;
-  private double timeInit;
+  private TimedBoolean isCurrentSpiking;
 
-  public PivotCalibrate(PivotIntake pivotIntake, int position) {
-    // position: 0 is ground, 1 is shooter
-    this.pivotIntake = pivotIntake;
+  // position: 0 is ground, 1 is shooter
```
Comment (3LucasZ) [inline src/main/java/frc/robot/subsystems/pivot/commands/PivotCalibrate.java:21]: change to boolean (isShooter)

File: src/main/java/frc/robot/subsystems/pivot/commands/PivotCalibrate.java [hunk 3]
```diff
@@ -7,64 +7,66 @@
...
+      throw new IllegalArgumentException("Invalid position: " + position);
+    }
+    if (FeatureFlags.kDebugEnabled) {
+      System.out.println("PivotSlam: " + position);
+    }
+    this.pivot = pivot;
     this.position = position;
-    addRequirements(pivotIntake);
+    this.isCurrentSpiking =
+        new TimedBoolean(pivot::isCurrentSpiking, 0.5); // Have we been spiking for 0.5 seconds?
```
Comment (3LucasZ) [inline src/main/java/frc/robot/subsystems/pivot/commands/PivotCalibrate.java:32]: The spike happens very fast, just check isCurrentSpiking without TimedBoolean. 0.5s is too long too.

**Review by 3LucasZ — CHANGES_REQUESTED** [review]
Not done with my PR request. If someone can take over from here on out please do.

**Pull request merged**